plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/SwitchGroup/hooks.ts

```diff
@@ -0,0 +1,124 @@
+import { array_move } from "@web/extensions/sidebar/utils";
+import { useCallback, useEffect, useState } from "react";
+
+import { groupItem, SwitchGroup } from "../../types";
+
+export default (value: SwitchGroup) => {
+  const [switchGroupObj, updateGroups] = useState<SwitchGroup>(value);
+  const [groupsTitle, setGroupsTitle] = useState(value.title);
+  const [currentGroup, setCurrentGroup] = useState<groupItem>(value.groups[0]);
+  const [modifiedGroups, updateModifiedGroups] = useState<SwitchGroup | undefined>();
+
+  //initialize the helper array (modifiedGroups)
+  useEffect(() => {
+    updateModifiedGroups({ type: "switchGroup", title: "Temp", groups: [] });
+  }, []);
+
+  //add empty item each time we press on add item
+  const handAddItem = useCallback(() => {
+    updateModifiedGroups(l => {
+      if (!l) return;
+      l.groups.push({ id: 0, group: "", title: "" });
+      return { ...l, groups: l.groups };
+    });
+  }, []);
+
+  //modiy the group in the helper array
+  const handleModifyGroup = (group: string, index: number) => {
+    updateModifiedGroups(l => {
+      let newArray: groupItem[] | undefined = undefined;
+      if (!l || !l.groups) return;
```

Comment:
You can just do `!l?.groups`.
Also realize, I think you copy/pasted from the Legend component, right? The reason it is `l` is because Legend starts with L. Please make sure to update copy/pasted code to represent where it is being used.